simsity/service.py

```diff
@@ -34,10 +38,21 @@ def train_from_dataf(self, df, features=None):
         subset = df
         if features:
             subset = df[features]
+
         self.storage = {i: r for i, r in enumerate(subset.to_dict(orient="records"))}
-        data = self.encoder.fit_transform(subset)
+        
+        if self._trained:
+            try:
+                data = self.encoder.transform(subset)
+            except Exception as e:
+                warnings.warn("Encountered error using pretrained encoder. Are you sure it is trained?")
+                raise e
+        else:
+            data = self.encoder.fit_transform(subset)
```

Comment:
I think we're about to omit another `else` statement here to keep things succinct. 

```python
if not self._trained:
    self.encoder.fit(subset)

try:
    data = self.encoder.transform(subset)
except Exception as e:
    warnings.warn("Encountered error using pretrained encoder. Are you sure it is trained?")
    raise e
```